google-cloud-bigquerystorage/src/test/java/com/google/cloud/bigquery/storage/v1/it/ITBigQueryWriteManualClientTest.java

```diff
@@ -468,6 +433,172 @@ public void testJsonStreamWriterWithDefaultSchema()
+              ByteString.copyFromUtf8("a").toByteArray(), ByteString.copyFromUtf8("b").toByteArray()
+            }));
+    row1.put("test_timestamp", "2022-02-06 07:24:47.84");
+    JSONObject row2 = new JSONObject();
+    row2.put("test_str", "bbb");
+    row2.put(
+        "test_bytestring_repeated",
+        new JSONArray(
+            new byte[][] {
+              ByteString.copyFromUtf8("c").toByteArray(), ByteString.copyFromUtf8("d").toByteArray()
+            }));
+    JSONObject row3 = new JSONObject();
+    row3.put("test_str", "ccc");
+    JSONObject row4 = new JSONObject();
+    row4.put("test_str", "ddd");
+    JSONObject row5 = new JSONObject();
+    // Add another ARRAY<BYTES> using a more idiomatic way
+    JSONArray testArr = new JSONArray(); // create empty JSONArray
+    testArr.put(0, ByteString.copyFromUtf8("a").toByteArray()); // insert 1st bytes array
+    testArr.put(1, ByteString.copyFromUtf8("b").toByteArray()); // insert 2nd bytes array
+    row5.put("test_bytestring_repeated", testArr);
+
+    JSONArray jsonArr1 = new JSONArray(new JSONObject[] {row1});
+    JSONArray jsonArr2 = new JSONArray();
+    jsonArr2.put(row2);
+    jsonArr2.put(row3);
+    JSONArray jsonArr3 = new JSONArray();
+    jsonArr3.put(row4);
+    JSONArray jsonArr4 = new JSONArray();
+    jsonArr4.put(row5);
+
+    bigquery.create(tableInfo);
+    TableName parent = TableName.of(ServiceOptions.getDefaultProjectId(), DATASET, tableName);
+    try {
+      // JsonStream write created without TableSchema
+      JsonStreamWriter jsonStreamWriter1 =
+          JsonStreamWriter.newBuilder(parent.toString(), client)
+              .setIgnoreUnknownFields(true)
+              .build();
+      // JsonStream write created with TableSchema
+      JsonStreamWriter jsonStreamWriter2 =
+          JsonStreamWriter.newBuilder(parent.toString(), tableSchema)
+              .setIgnoreUnknownFields(true)
+              .build();
+
+      assertEquals(jsonStreamWriter1.getWriterId(), jsonStreamWriter2.getWriterId());
+      LOG.info("Sending one message");
+      ApiFuture<AppendRowsResponse> response1 = jsonStreamWriter1.append(jsonArr1, -1);
+      LOG.info("Sending three more messages");
+      ApiFuture<AppendRowsResponse> response2 = jsonStreamWriter2.append(jsonArr2, -1);
+      LOG.info("Sending two more messages");
+      ApiFuture<AppendRowsResponse> response3 = jsonStreamWriter2.append(jsonArr3, -1);
+      LOG.info("Sending one more message");
+      ApiFuture<AppendRowsResponse> response4 = jsonStreamWriter1.append(jsonArr4, -1);
+
+      Assert.assertFalse(response1.get().getAppendResult().hasOffset());
+      Assert.assertFalse(response2.get().getAppendResult().hasOffset());
+      Assert.assertFalse(response3.get().getAppendResult().hasOffset());
+      Assert.assertFalse(response4.get().getAppendResult().hasOffset());
+
+      LOG.info("Both JsonStreamWriters wrote to the same Table successfully!");
+    } catch (DescriptorValidationException e) {
```

Comment:
Will the test be passing with these exceptions? We should fail the test.

Reply:
Added test for failure.